File: html/cgi/python/LogOut.py
import os
import datetime


def logOut():
    print("Content-type: text/html")
    body = "<!DOCTYPE html>"
    body += "<html><head><title>Bye Bye!</title>"
    body += "</title><link rel=\"icon\" type=\"image/x-icon\" href=\"py.jpg\">"
    body += "<style>* "
    body += "{font-family: system-ui, -apple-system, BlinkMacSystemFont,"
    body += "'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', "
    body += "'Helvetica Neue', sans-serif;} body { background-color: "
    body += "CornflowerBlue; display: flex; justify-content: center; "
    body += "align-items: center; height: 100vh; margin: 0; flex-direction: "
    body += "column;} h1 { font-size: 3em; color: black; } </style></head>"
    body += "<body><h1>See Ya!</h1>"
    body += "<br><a href=\"/python/welcome.py\"><button>Go back to main page"
    body += "</button></a></body></html>"
    if "SERVER_NAME" in os.environ:
        serverName = os.environ.get('SERVER_NAME')
        print(f"Server: {serverName}")
    datetimenow = datetime.datetime.now()
    date = datetimenow.astimezone().strftime("%a, %d %b %Y %H:%M:%S %Z")
    print(f"Date: {date}")
    time = "Thu, 01-Jan-1970 00:00:10 GMT"
    print(f"Set-Cookie: user=blbkvvv;Expires={time};")
    print(f"Set-Cookie: bgcolor=spspspsp;Expires={time};")
    print(f"Set-Cookie: python_img=sasvcb;Expires={time};")
    print(f"Content-Length: {len(body)}\r\n")
    print(f"{body}")


if __name__ == "__main__":
    logOut()

File: html/cgi/ruby/LogOut.rb
require 'time'

def log_out
  # Prepare the HTML body
  body = <<~HTML
    <!DOCTYPE html>
    <html><head><title>Bye Bye!</title>
    <style>* {
    font-family: system-ui, -apple-system, BlinkMacSystemFont,
    'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans',
    'Helvetica Neue', sans-serif;} body { background-color:
    CornflowerBlue; display: flex; justify-content: center;
    align-items: center; height: 100vh; margin: 0; flex-direction:
    column;} h1 { font-size: 3em; color: black; } </style></head>
    <body><h1>See Ya!</h1>
    <br><a href="/ruby/welcome.rb"><button>Go back to main page</button></a></body></html>
  HTML

  # Print HTTP headers
  puts "Content-Type: text/html"
  puts "Set-Cookie: user=blbkvvv; Expires=Thu, 01-Jan-1970 00:00:10 GMT"
  puts "Set-Cookie: bgcolor=spspspsp; Expires=Thu, 01-Jan-1970 00:00:10 GMT"

  # Print server name if available
  if ENV['SERVER_NAME']
    server_name = ENV['SERVER_NAME']
    puts "Server: #{server_name}"
  end

  # Print current date
  date = Time.now.strftime("%a, %d %b %Y %H:%M:%S %Z")
  puts "Date: #{date}"

  # Print content length and body
  puts "Content-Length: #{body.bytesize}\r\n"
  puts "\r\n"
  puts body
end

if __FILE__ == $0
  log_out
end
